crates/client/rpc/src/trace_api.rs

```diff
@@ -186,12 +192,18 @@ fn try_get_funtion_invocation_from_call_info<B: BlockT>(
     // address". We have to do this decution ourselves here
     let class_hash = if let Some(class_hash) = call_info.call.class_hash {
         class_hash.0.into()
+    } else if let Some(cached_hash) = class_hash_cache.get(&call_info.call.storage_address) {
+        *cached_hash
     } else {
-        let class_hash = storage_override
+        // Compute and cache the class hash
+        let computed_hash = storage_override
             .contract_class_hash_by_address(substrate_block_hash, call_info.call.storage_address)
             .ok_or_else(|| TryFuntionInvocationFromCallInfoError::ContractNotFound)?;
 
-        FieldElement::from_byte_slice_be(class_hash.0.bytes()).unwrap()
+        let field_element = FieldElement::from_byte_slice_be(computed_hash.0.bytes()).unwrap();
```

Comment:
Rename to `computed_hash` rather than `field_element`, that will shadow the previous `computed_hash`